src/mgclient.c

```diff
@@ -688,6 +705,10 @@ int mg_session_run(mg_session *session, const char *query, const mg_map *params,
     goto fatal_failure;
   }
 
+#ifdef __EMSCRIPTEN_
+  int socket = ((mg_raw_transport *)session->transport)->sockfd;
+  yield_until_async_read(socket, 10);
+#endif
```

Comment:
I don't like this cast here. What about handling it similarly to `mg_raw_transport_send`, so storing a pointer to the function in the transport object.